tests/components/fan/test_template.py

```diff
@@ -0,0 +1,536 @@
+"""The tests for the Template fan platform."""
+import logging
+
+from homeassistant.core import callback
+from homeassistant import setup
+import homeassistant.components as components
+from homeassistant.const import STATE_ON, STATE_OFF
+from homeassistant.components.fan import (
+    ATTR_SPEED, ATTR_OSCILLATING, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH)
+
+from tests.common import (
+    get_test_home_assistant, assert_setup_component)
+_LOGGER = logging.getLogger(__name__)
+
+
+_TEST_FAN = 'fan.test_fan'
+# Represent for fan's state
+_STATE_INPUT_BOOLEAN = 'input_boolean.state'
+# Represent for fan's speed
+_SPEED_INPUT_SELECT = 'input_select.speed'
+# Represent for fan's oscillating
+_OSC_INPUT_SELECT = 'input_select.osc'
+
+
+class TestTemplateFan:
+    """Test the Template light."""
+
+    hass = None
+    calls = None
+    # pylint: disable=invalid-name
+
+    def setup_method(self, method):
+        """Setup."""
+        self.hass = get_test_home_assistant()
+
+        self.calls = []
+
+        @callback
+        def record_call(service):
+            """Track function calls.."""
+            self.calls.append(service)
+
+        self.hass.services.register('test', 'automation', record_call)
+
+    def teardown_method(self, method):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    # Configuration tests #
+    def test_missing_optional_config(self):
+        """Test: missing optional template is ok."""
+        with assert_setup_component(1, 'fan'):
+            assert setup.setup_component(self.hass, 'fan', {
+                'fan': {
+                    'platform': 'template',
+                    'fans': {
+                        'test_fan': {
+                            'value_template': "{{ 'on' }}",
+
+                            'turn_on': {
+                                'service': 'script.fan_on'
+                            },
+                            'turn_off': {
+                                'service': 'script.fan_off'
+                            }
+                        }
+                    }
+                }
+            })
+
+        self.hass.start()
```

Comment:
No need to call `self.hass.start()` in any of these test cases.

Reply:
removing them failed the tests.